sqlitecluster/SQLiteNode.cpp

```diff
@@ -1779,14 +1924,36 @@ void SQLiteNode::_changeState(SQLiteNode::State newState) {
     // Exclusively lock the stateMutex, nobody else will be able to get a shared lock until this is released.
     unique_lock<decltype(stateMutex)> lock(stateMutex);
 
-    // We send any unsent transactions here before we finish switching states. Normally, this does nothing, unless
-    // we're switching down from LEADING or STANDINGDOWN, but we need to make sure these are all sent to the new
-    // leader before we complete the transition.
-    _sendOutstandingTransactions();
-
     // Did we actually change _state?
     State oldState = _state;
+
+    // Not sure if this is entirely adequate.
+    SINFO("[NOTIFY] setting commit count to: " << _db.getCommitCount());
+    _dbNotifier.notifyThrough(_db.getCommitCount());
+
     if (newState != oldState) {
+        // If we were following, and now we're not, we give up an any replications.
+        if (oldState == FOLLOWING) {
+            _replicationThreadsShouldExit = true;
+            SINFO("[NOTIFY] _replicationThreadsShouldExit");
+            _dbNotifier.cancel();
+            _commitNotifier.cancel();
+
+            // Polling wait for threads to quit. This could use a notification model such as with a condition_variable,
+            // which would probably be "better" but introduces yet more state variables for a state that we're rarely
+            // in, and so I've left it out for the time being.
+            while (_replicationThreadCount) {
+                usleep(10'000);
```

Comment:
Does this wait 10s per loop?  Can we wait like 1s, or even 100ms?

Reply:
It waits 10,000 microseconds (10ms).